src/android/BluetoothLePlugin.java

```diff
@@ -2905,6 +2905,9 @@ public void onScanResult(int callbackType, ScanResult result) {
           JSONObject returnObj = new JSONObject();
 
           addDevice(returnObj, result.getDevice());
+          if(result.getScanRecord().getDeviceName() != null){
+            addProperty(returnObj, keyName, result.getScanRecord().getDeviceName().replace("\0", ""));
```

Comment:
If this is dependent on a specific device, I think it should be done on the client side.